bindings/node.js/README.md

```diff
@@ -1,93 +1,146 @@
+- A compiler like g++ or clang
+- [`make`](https://www.gnu.org/software/make/manual/make.html)
+- [`python3`](https://docs.python.org/3/)
+
+## Installation
 
 ```sh
 yarn add c-kzg
+# or
+npm i -S c-kzg
 ```
 
-Import from it like any other library
+## Usage
 
-```js
+```ts
 import {
-  KZGCommitment,
+  BYTES_PER_BLOB,
+  Blob,
+  Bytes48,
   blobToKzgCommitment,
-  verifyAggregateKzgProof,
-  loadTrustedSetup,
-  transformTrustedSetupJSON,
+  computeBlobKzgProof,
+  verifyBlobKzgProofBatch,
 } from "c-kzg";
-```
-
-# Requirements
-
-The C and C++ code is compiled by node-gyp on installation. Your environment will need
-
-- A compiler like g++ or clang
-- `make`
-- `python3`
-
-# Contributing
 
-This directory contains the code necessary to generate NodeJS bindings for C-KZG.
-
-The bindings file has the following interface:
-
-```js
-
-  loadTrustedSetup: (filePath: string) => SetupHandle;
-
-  freeTrustedSetup: (setupHandle: SetupHandle) => void;
-
-  blobToKzgCommitment: (blob: Blob, setupHandle: SetupHandle) => KZGCommitment;
-
-  computeAggregateKzgProof: (
-    blobs: Blob[],
-    setupHandle: SetupHandle
-  ) => KZGProof;
-
-  verifyAggregateKzgProof: (
-    blobs: Blob[],
-    expectedKzgCommitments: KZGCommitment[],
-    kzgAggregatedProof: KZGProof,
-    setupHandle: SetupHandle
-  ) => boolean;
+const blobs = [] as Blob[];
+const commitments = [] as Bytes48[];
+const proofs = [] as Bytes48[];
+
+for (let i = 0; i < BATCH_SIZE; i++) {
+  blobs.push(
+    Buffer.alloc(BYTES_PER_BLOB, "*"));
+  commitments.push(
+    blobToKzgCommitment(blobs[i]));
+  proofs.push(
+    computeBlobKzgProof(blobs[i], commitments[i]));
+  // note: blobs[i] relates to commitments[i] and proofs[i]
+  //       when running `verifyBlobKzgProofBatch`
+}
+
+const isValid = verifyBlobKzgProofBatch(blobs, commitments, proofs);
 ```
 
-But this library wraps it in module with manages the setupHandle internally.
-
-First,
-`npm install -g yarn` if you don't have it.
-
-Install the blst submodule
-
-```sh
-git submodule update --init
-```
-
-Build blst and c_kzg_4844.c
-
-```
-cd src && make blst lib
-```
-
-Generate NodeJS bindings and run the TypeScript tests against them
-
-```sh
-cd ../bindings/node.js && yarn install && make test
-```
-
-After doing this once, you can re-build (if necessary) and re-run the tests with
-
-```sh
-make build test
-```
-
-After making changes, regenerate the distributable JS and type defs
-
-```sh
-make bundle
+## API
+
+```ts
+/**
+ * Sets up the c-kzg library. Pass in a properly formatted trusted setup file
+ * to configure the library.  File must be in json format, see TrustedSetupJson
+ * interface for more details, or as a properly formatted utf-8 encoded file.
+ *
+ * @remark This function must be run before any other functions in this
+ *         library can be run.
+ *
+ * @param {string} filePath - The absolute path of the trusted setup
+ */
+export function loadTrustedSetup(filePath: string): void;
```

Comment:
I think this would look better if each function were split into its own section. Like:

## API
### `loadTrustedSetup`

```
<func definition>
```

### `blobToKzgCommitment`

```
<func definition>
```

...

Reply:
Ok. Updated.  I moved the whole doc string with param and return info.  I think that is what you were referring to but will be happy to remove everything but the function signature